src/link/updateShrinkwrap.ts

```diff
@@ -105,10 +105,18 @@ function updateResolvedDeps (
 }
 
 function toShrResolution (dependencyPath: string, resolution: Resolution): ShrinkwrapResolution {
-  if (!dp.isAbsolute(dependencyPath) && resolution.type === undefined && resolution.integrity) {
+  if (dp.isAbsolute(dependencyPath) || resolution.type !== undefined || !resolution.integrity) {
+    return resolution
+  }
+  // This might be not the best solution to identify non-standard tarball URLs in the long run
+  // but it at least solves the issues with npm Enterprise. See https://github.com/pnpm/pnpm/issues/867
+  if (!resolution.tarball.includes('/-/')) {
```

Comment:
FWIW, we have an email out to NPM asking them about the URL schemes for NPM Enterprise. We'll let you know if we learn anything that could make this test better.